pkg/image/util/signature/signature.go

```diff
@@ -0,0 +1,65 @@
+package signature
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"time"
+
+	_ "github.com/containers/image/docker"
+	"github.com/containers/image/transports"
+	"github.com/containers/image/types"
+
+	imageapi "github.com/openshift/origin/pkg/image/apis/image"
+)
+
+// defaultTransportName is the name of default containers/image transport to use
+// to pull signatures.
+const defaultTransportName = "docker"
+
+// defaultGetSignatureTimeout represents a maximum number of seconds that we are
+// willing to wait to get the signature from remote signature store.
+var defaultGetSignatureTimeout = 20 * time.Second
+
+// GetSignatureForImage pulls down image signatures from remote or local signature store.
+// See this document for more details about how to configure the signature stores:
+// https://github.com/containers/image/blob/master/docs/registries.d.md
+func GetSignatureForImage(registryHost, repositoryName, imageName string) ([]imageapi.ImageSignature, error) {
+	t := transports.Get(defaultTransportName)
+	reference, err := t.ParseReference("//" + registryHost + "/" + repositoryName + "@" + imageName)
+	if err != nil {
+		return nil, err
+	}
+	source, err := reference.NewImageSource(&types.SystemContext{}, nil)
```

Comment:
@mtrmac also dealing with authz here will add yet another level of complexity to this code. I will have to pass in the proper secrets/etc.